staging/src/k8s.io/apiserver/pkg/endpoints/request/requestcontext.go

```diff
@@ -40,37 +41,55 @@ type RequestContextMapper interface {
 }
 
 type requestContextMap struct {
-	contexts map[*http.Request]Context
-	lock     sync.Mutex
+	contexts map[*http.Request]*atomic.Value
+	lock     sync.RWMutex
 }
 
 // NewRequestContextMapper returns a new RequestContextMapper.
 // The returned mapper must be added as a request filter using NewRequestContextFilter.
 func NewRequestContextMapper() RequestContextMapper {
 	return &requestContextMap{
-		contexts: make(map[*http.Request]Context),
+		contexts: make(map[*http.Request]*atomic.Value),
 	}
 }
 
+func (c *requestContextMap) getValue(req *http.Request) (*atomic.Value, bool) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	value, ok := c.contexts[req]
+	return value, ok
+}
+
+type contextWrap struct {
```

Comment:
why this wrapper? This needs a godoc.

Reply:
to prevent atomic.Value.Store() panic  
https://github.com/kubernetes/kubernetes/pull/60824#discussion_r172734694

Reply:
done